File: Server/ServerCShop0/ServerCShop0/Protocol.cs
namespace Protocol
{
    public enum EnumKey : ushort
    {
        reqLogin = 0,
        resLogin = 1,
        reqRoomAreaJoin = 2,
        resRoomAreaJoin = 3,
        reqRoomAreaMessage = 4,
        reqMessage = 10,
        resMessage = 11,
    }
    public enum EnumResResult : byte
    {
        SUCCESS = 0,
    }

    public class BasePacket
    {
        public EnumKey key { get; set; }
    }
    public class BasePacketRes : BasePacket
    {
        public EnumResResult result { get; set; }
    }

    public class ReqLogin : BasePacket{
        public string nick { get; set; }
    }
    public class ResLogin : ResRoomAreaList
    {
        
    }
    public class ReqRoomAreaJoin : BasePacket
    {
        public string idRoom { get; set; }
    }
    public class ResRoomAreaJoin : BasePacketRes
    {

    }


    public class ResRoomAreaList : BasePacketRes
    {
        public RecordRoomArea[] recordRoomAreas { get; set; }
    }
    public class RecordRoomArea
    {
        public string idRoom { get; set; }
        public string name { get; set; }
    }

    public class ReqRoomAreaMessage : BasePacket
    {
        public string idRoom { get; set; }
        public string message { get; set; }
    }

    public class ReqMessage : BasePacket
    {
        public string message { get; set; }
    }
    public class ResMessage : BasePacketRes
    {
        public string message { get; set; }
    }
}

File: Server/WorkspaceJava/ServerJava0/src/com/server/Protocol.java
package com.server;

public class Protocol {
	public enum EnumKey
    {
        reqLogin(0),
        resLogin(1),
        reqRoomAreaJoin(2),
        resRoomAreaJoin(3),
        reqRoomAreaMessage(4),
        reqMessage(10),
        resMessage(11);
        
        final private int name; 
        private EnumKey(int name) { 
        	this.name = name; 
        } 
        public int getName() {
        	return name; 
        }
    }
	
	//for(Type type : Type.values()){ System.out.println(type.getName()); }

	
    public enum EnumResResult
    {
        SUCCESS(0);
        final private int name; 
        private EnumResResult(int name) { 
        	this.name = name; 
        } 
        public int getName() {
        	return name; 
        }
    }

    
}

class BasePacket
{
    public Protocol.EnumKey key;
}
class BasePacketRes extends BasePacket
{
    public Protocol.EnumResResult result;
}

 class ReqLogin extends BasePacket
{
    public String nick;
}
 class ResLogin extends ResRoomAreaList
{

}
 class ReqRoomAreaJoin extends BasePacket
{
    public String idRoom;
}
 class ResRoomAreaJoin extends BasePacketRes
{

}


 class ResRoomAreaList extends BasePacketRes
{
    public RecordRoomArea[] recordRoomAreas;
}
 class RecordRoomArea
{
    public String idRoom;
    public String name;
}

 class ReqRoomAreaMessage extends BasePacket
{
    public String idRoom;
    public String message;
}

 class ReqMessage extends BasePacket
{
    public String message;
}
 class ResMessage extends BasePacketRes
{
    public String message;
}